src/tasks/netcore/updateBlazorManifest.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as xml2js from 'xml2js';
+import ChildProcessProvider from "../../debugging/coreclr/ChildProcessProvider";
+import { DockerContainerVolume } from '../../debugging/coreclr/CliDockerClient';
+import LocalOSProvider from "../../debugging/coreclr/LocalOSProvider";
+import { OSTempFileProvider } from "../../debugging/coreclr/tempFileProvider";
+import { ext } from "../../extensionVariables";
+import { pathNormalize } from '../../utils/pathNormalize';
+import { PlatformOS } from '../../utils/platform';
+import { execAsync } from '../../utils/spawnAsync';
+import { DockerRunTaskDefinition } from "../DockerRunTaskProvider";
+import { DockerRunTaskContext } from "../TaskHelper";
+
+interface ContentRootAttributes {
+    BasePath: string;
+    Path: string;
+}
+
+interface ContentRoot {
+    $: ContentRootAttributes;
+}
+
+interface StaticWebAssets {
+    ContentRoot: ContentRoot[];
+}
+
+interface Manifest {
+    StaticWebAssets: StaticWebAssets;
+}
+
+export async function updateBlazorManifest(context: DockerRunTaskContext, runDefinition: DockerRunTaskDefinition): Promise<void> {
+    const tempFileProvider = new OSTempFileProvider(new LocalOSProvider(), new ChildProcessProvider());
+
+    const locationsFile = tempFileProvider.getTempFilename();
+
+    const targetsFile = path.join(ext.context.asAbsolutePath('resources'), 'GetBlazorManifestLocations.targets');
+
+    const command = `dotnet build /r:false /t:GetBlazorManifestLocations /p:CustomAfterMicrosoftCommonTargets=${targetsFile} /p:BlazorManifestLocationsOutput=${locationsFile} ${runDefinition.netCore.appProject}`;
+
+    try {
+        await execAsync(command, { timeout: 5000 });
```

Comment:
Are you sure 5 seconds is enough?

Reply:
The Blazor thing is best-effort. Since we have no choice but to run it on nearly everything (see above comment), it would be better to timeout and kill the process than to take longer, since the projects actually needing it are probably a pretty small minority right now.

Additionally, the only way this takes longer than 5s would be if it's busted (in which case, we *should* terminate), or there's an absolutely massive amount of props/targets to evaluate.

In practice this only took ~300ms to run on a normal project, I think 5s is plenty.

Reply:
I think what you have is probably fine as a starting point; but I have seen cases when things completed within hundreds of milliseconds on my dev machine (not much else going on, content already cached in filesystem cache) vs took many seconds on a slow machine that just happened to do an anti-virus scan at the same time, completely saturating the storage subsystem.

We might consider augmenting our `execAsync` and `spawnAsync` methods to explicitly check for timeout condition and emit telemetry when that happens (that would be a separate PR of course), so that we know if this is a problem or not.

Reply:
That makes sense. We can also monitor debugging success rates, errors, and other things to try to get hints to see if timeouts are a common occurrence.